extension/src/format/parser.ts

```diff
@@ -120,7 +120,7 @@ export class LispParser {
                     endPos2d = document.positionAt(endOfString.offsetInDocument);
                 }
                 else {
-                    endPos2d = selectionStartOffset + textString.length;
+                    endPos2d = selectionStartOffset + textString.length; // BUG: Doesn't produce Position Object
```

Comment:
If you could save me the time of deciphering all of this and just give me some idea of what this probably should be returning, I'll fix/test it. I only ever hit this bug only during extremely malformed requests. IE, the document was the constructor context, but I only asked it to tokenize a single (possibly nested) defun; made some epic bad decisions with those two very different context points. And why I wound up generating a ReadonlyDocument for my smaller scope of work.

Reply:
I think this position is the index to treat the multiline text as plain text. So this is a index for whole document.
@Sen-real can correct me at this point.

Reply:
@hualin-adsk: I agree with @JD-Howard, it seems wrong. It's used to create a `vscode.Range `object, which should be created with either 2 "`vscode.Position`" or 4 "`Number`".
For Syntax, I think it should be `document.positionAt(selectionStartOffset + textString.length)`; but from logic point of view I can't tell if it produces expected result or not.

Reply:
I already know of a couple documents and/or bad requests that trigger this. So, I'll try the document.positionAt() and see if it returns something sensible.

Reply:
Cool!